scripts/insert_fake_data_and_test.py

```diff
@@ -0,0 +1,155 @@
+        {'a_id': 1, 't_id': 3},
+        {'a_id': 2, 't_id': 3},
+        {'a_id': 2, 't_id': 4},
+        {'a_id': 2, 't_id': 5},
+        {'a_id': 2, 't_id': 6},
+    ]
+
+    test_hypernyms = [
+        {'c_id': 1, 'd_id': 1},
+        {'c_id': 1, 'd_id': 2},
+        {'c_id': 1, 'd_id': 3},
+        {'c_id': 1, 'd_id': 4},
+        {'c_id': 1, 'd_id': 5},
+        {'c_id': 2, 'd_id': 1},
+        {'c_id': 2, 'd_id': 2},
+        {'c_id': 2, 'd_id': 3},
+        {'c_id': 2, 'd_id': 6},
+    ]
+
+    insert_batch(Article, test_articles)
+    insert_batch(Type, test_types)
+    insert_batch(WikiClass, test_classes)
+    insert_batch(DbpediaClass, test_dbpedia_classes)
+
+    insert_batch(ArticleType, test_article_types)
+    insert_batch(ArticleClass, test_article_classes)
+    insert_batch(Hypernym, test_hypernyms)
+
+
+class TestRelationships(unittest.TestCase):
+
+    def test_articles_of_class(self):
+        self.assertItemsEqual(["Paris", "London"],
+                              query.articles_of_class("wikipedia-city"))
+
+        # we're flexible in not requiring 'wikipedia-'
+        self.assertItemsEqual(["Paris"],
+                              query.articles_of_class("french commune"))
+
+    def test_classes_of_article(self):
+        self.assertItemsEqual(["wikipedia-president"],
+                              query.classes_of_article("Bill Clinton"))
+        self.assertItemsEqual(["wikipedia-city", "wikipedia-french-commune"],
+                              query.classes_of_article("Paris"))
+
+    def test_classes_of_article_with_limit(self):
+        self.assertEquals(1, len(query.classes_of_article("Paris", limit=1)))
+
+    def test_articles_of_type(self):
+        self.assertItemsEqual(["Bill Clinton", "Barack Obama"],
+                              query.articles_of_type("american"))
+
+    def test_articles_with_multiple_types(self):
+        articles = query.articles_with_multiple_types("american", "current president", op='and')
+        self.assertItemsEqual(["Barack Obama"], articles)
+
+        articles = query.articles_with_multiple_types("politician", "president", op='or')
+        self.assertItemsEqual(["Barack Obama", "Bill Clinton"], articles)
+
+    def test_types_of_article(self):
+        self.assertItemsEqual(
+            ["politician", "american politician", "american"],
+            query.types_of_article("Bill Clinton"))
+
+        self.assertItemsEqual(
+            ["american", "president", "44th president", "current president"],
+            query.types_of_article("Barack Obama"))
+
+    def test_hypernyms_of_article_from_db(self):
+        president_hypernyms = ['president', 'politician', 'person', 'agent', 'thing']
+        actual_hypernyms = query.hypernyms_of_article_from_db("Bill Clinton")
+        self.assertItemsEqual(actual_hypernyms, president_hypernyms)
+
+    @unittest.expectedFailure
+    def test_articles_of_hypernym_from_db(self):
+        # TODO: wrong hypernyms are being returned
```

Comment:
My guess is that the JOIN query has a bug. Since getting hypernyms from the DB is deprecated (and low priority), we should open an issue and fix it later.